extensions/sql-database-projects/src/dialogs/addSqlBindingQuickpick.ts

```diff
@@ -151,28 +153,29 @@ export async function launchAddSqlBindingQuickpick(uri: vscode.Uri | undefined,
 					continue;
 				}
 
-				const newConnectionStringValue = await vscode.window.showInputBox(
-					{
-						title: constants.enterConnectionString,
-						ignoreFocusOut: true,
-						value: 'Server=localhost;Initial Catalog={db_name};User ID=sa;Password={your_password};Persist Security Info=False',
-						validateInput: input => input ? undefined : constants.valueMustNotBeEmpty
-					}
-				) ?? '';
-
-				if (!newConnectionStringValue) {
-					// go back to select setting quickpick if user escapes from inputting the value in case they changed their mind
-					continue;
+				// get connection string for sql server
+				let connectionString: string = '';
+				const connectionInfo = await vscodeMssqlApi.promptForConnection(true);
+				if (!connectionInfo) {
+					// User cancelled
+					return;
 				}
-
+				const connectionUri = await vscodeMssqlApi.connect(connectionInfo);
 				try {
-					const success = await azureFunctionsUtils.setLocalAppSetting(path.dirname(projectUri.fsPath), newConnectionStringSettingName, newConnectionStringValue);
-					if (success) {
-						connectionStringSettingName = newConnectionStringSettingName;
+					connectionString = await vscodeMssqlApi.getConnectionString(connectionUri, false);
+					try {
+						const success = await azureFunctionsUtils.setLocalAppSetting(path.dirname(projectUri.fsPath), newConnectionStringSettingName, connectionString);
+						if (success) {
+							connectionStringSettingName = newConnectionStringSettingName;
+						}
+					} catch (e) {
+						// display error message and show select setting quickpick again
+						void vscode.window.showErrorMessage(utils.getErrorMessage(e));
 					}
 				} catch (e) {
-					// display error message and show select setting quickpick again
-					void vscode.window.showErrorMessage(utils.getErrorMessage(e));
+					// go back to select setting quickpick if user escapes from inputting the value in case they changed their mind
+					console.warn(e);
```

Comment:
We should display an error to the user so they know something went wrong. It likely isn't useful to display the actual error message so just something like `An error occurred generating the connection string for the selected connection` should suffice (but still log the error to the console)